File: apps/api/src/lib/llm.ts
import { anthropic } from "@ai-sdk/anthropic";
import type { LanguageModel } from "ai";

export function getModel(): LanguageModel {
  const provider = process.env.LLM_PROVIDER ?? "anthropic";
  const model = process.env.LLM_MODEL ?? "claude-sonnet-4-6";
  if (provider === "anthropic") return anthropic(model);
  // openai/google: add @ai-sdk/openai|google and branch here when toggled
  throw new Error(`Unsupported LLM_PROVIDER for resolver: ${provider}`);
}

File: apps/agent/app/llm.py
import os


def make_llm():
    provider = os.getenv("LLM_PROVIDER", "anthropic")
    model = os.getenv("LLM_MODEL", "claude-sonnet-4-6")
    if provider == "anthropic":
        from browser_use import ChatAnthropic
        return ChatAnthropic(model=model)
    raise ValueError(f"Unknown LLM_PROVIDER: {provider}")
